src/Motor.Extensions.Hosting.RabbitMQ/RabbitMQPriorityExtension.cs

```diff
@@ -1,78 +1,30 @@
+        public static CloudEventAttribute RabbitMQPriorityAttribute { get; } =
+            CloudEventAttribute.CreateExtension("priority", CloudEventAttributeType.Integer);
 
-        public RabbitMQPriorityExtension(byte priority)
-        {
-            Priority = priority;
-        }
-
-
-        public byte? Priority
-        {
-            get => (byte?)attributes[PriorityAttributeName];
-            set
-            {
-                if (value is not null) attributes[PriorityAttributeName] = value;
-            }
-        }
+        public static IEnumerable<CloudEventAttribute> AllAttributes { get; } =
+            new[] { RabbitMQPriorityAttribute }.ToList().AsReadOnly();
 
-        public void Attach(CloudEvent cloudEvent)
+        public static MotorCloudEvent<TData> SetRabbitMQPriority<TData>(this MotorCloudEvent<TData> cloudEvent, byte? value) where TData : class
         {
-            var eventAttributes = cloudEvent.GetAttributes();
-            if (attributes == eventAttributes)
-                // already done
-                return;
-
-            foreach (var attr in attributes) eventAttributes[attr.Key] = attr.Value;
-
-            attributes = eventAttributes;
-        }
-
-        public bool ValidateAndNormalize(string key, ref object value)
-        {
-            switch (key)
-            {
-                case PriorityAttributeName:
-                    switch (value)
-                    {
-                        case null:
-                            return true;
-                        case string s:
-                            {
-                                if (!byte.TryParse(s, out var i))
-                                    throw new InvalidOperationException("ErrorPriorityValueIsaNotAnInteger");
-                                value = (byte?)i;
-                                return true;
-                            }
-                        case byte b:
-                            value = b;
-                            return true;
-                        default:
-                            throw new InvalidOperationException("ErrorPriorityValueIsaNotAnInteger");
-                    }
-            }
-
-            return false;
+            Validation.CheckNotNull(cloudEvent, nameof(cloudEvent));
+            cloudEvent[RabbitMQPriorityAttribute] = (int?)value;
+            return cloudEvent;
         }
 
-        // Disabled null check because CloudEvent SDK doesn't 
-        // implement null-checks
-#pragma warning disable CS8603
-        public Type GetAttributeType(string name)
+        public static byte? GetRabbitMQPriority<TData>(this MotorCloudEvent<TData> cloudEvent) where TData : class
         {
-            return name switch
-            {
-                PriorityAttributeName => typeof(byte?),
-                _ => null
-            };
+            var priority = (int?)Validation.CheckNotNull(cloudEvent, nameof(cloudEvent))[RabbitMQPriorityAttribute];
```

Comment:
nit: pattern matching ftw:

```cs
        public static byte? GetRabbitMQPriority<TData>(this MotorCloudEvent<TData> cloudEvent) where TData : class
        {
            return Validation.CheckNotNull(cloudEvent, nameof(cloudEvent))[RabbitMQPriorityAttribute] switch
            {
                int and (< 0 or > 255) => null,
                int priority => (byte) priority,
                _ => null
            };
        }
```

Reply:
I really still have to get used to pattern matching, but I definitely like this :+1: